### PR TITLE
Add multiple password prompt when importing p12 keys

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
Related to http://teampulse.telerik.com/view#item/285750

### CODE REVIEW

File: lib/commands/cryptographic-identities.ts [hunk 1]
```diff
@@ -602,11 +604,32 @@ export class ImportCryptographicIdentity implements ICommand {
...
+				targetFile = this.$fs.createReadStream(certificateFile);
+				if (isPasswordRequired) {
+					password = this.$prompter.getPassword("Certificate file password", { allowEmpty: true }).wait();
+				}
+
+				try {
+					result = this.$server.identityStore.importIdentity(<any>importType, password, targetFile).wait();
+				}
+				catch(err) {
+					noErrorOccurred = false;
```
Comment (rosen-vladimirov) [inline lib/commands/cryptographic-identities.ts]: what will happen if the problem with the import is not from the password? will you still ask for password 3 times?

File: lib/commands/cryptographic-identities.ts [hunk 2]
```diff
@@ -602,11 +604,32 @@ export class ImportCryptographicIdentity implements ICommand {
...
+					this.$logger.error(err.message);
+				}
+
+				if (noErrorOccurred) {
+					break;
+				}
+			}
```
Comment (rosen-vladimirov) [inline lib/commands/cryptographic-identities.ts]: Just one idea - if the three calls fail, the code continues. Currently this is not a problem, but maybe it is a good idea to call $errors.fail (...) in such case

File: lib/commands/cryptographic-identities.ts [hunk 3]
```diff
@@ -602,11 +604,32 @@ export class ImportCryptographicIdentity implements ICommand {
...
+				
+				targetFile = this.$fs.createReadStream(certificateFile);
+				if (isPasswordRequired) {
+					password = this.$prompter.getPassword("Certificate file password", { allowEmpty: true }).wait();
+				}
+
+				try {
+					result = this.$server.identityStore.importIdentity(<any>importType, password, targetFile).wait();
+				}
+				catch(err) {
```
Comment (rosen-vladimirov) [inline lib/commands/cryptographic-identities.ts]: can you move catch on the line above

File: lib/commands/cryptographic-identities.ts [hunk 4]
```diff
@@ -602,11 +604,35 @@ export class ImportCryptographicIdentity implements ICommand {
...
+					this.$logger.error(err.message + " Verify that you are entering the correct file and password.");
+				}
+
+				if (noErrorOccurred) {
+					break;
+				}
+			}
+
+			if (!noErrorOccurred) {
+				this.$errors.failWithoutHelp("Exceeded maximuim number of import attempts.");
```
Comment (rosen-vladimirov) [inline lib/commands/cryptographic-identities.ts]: ping @ikoevska  for the error message

Comment (ikoevska) [inline lib/commands/cryptographic-identities.ts]: Cannot import file. Verify that you have provided the correct file and password and try again.

You have reached the maximum number of authentication attempts for this operation.

**Comment by Mitko-Kerezov** [comment]
Comments by @ikoevska and @rosen-vladimirov addressed

**Comment by rosen-vladimirov** [comment]
:+1:

**Pull request merged**